cob_cartesian_controller/src/cartesian_controller.cpp

```diff
@@ -29,430 +29,366 @@
 #include <algorithm>
 
 #include <ros/ros.h>
-#include "ros/package.h"
 #include <std_srvs/Empty.h>
 #include <cob_srvs/SetString.h>
-#include <kdl_conversions/kdl_msg.h>
 
 #include <cob_cartesian_controller/cartesian_controller.h>
-#include <cob_cartesian_controller/trajectory_interpolator/trajectory_interpolator.h>
 
 bool CartesianController::initialize()
 {
     ros::NodeHandle nh_private("~");
 
-    ///get params articulation Nodehandle
-    if(!nh_private.getParam("reference_frame", reference_frame_))
+    if (!nh_.getParam("chain_tip_link", chain_tip_link_))
     {
-        ROS_ERROR("Parameter 'reference_frame' not set");
+        ROS_ERROR("Parameter 'chain_tip_link' not set");
         return false;
     }
-
-    if(!nh_private.getParam("target_frame", target_frame_))
+    
+    if(!nh_.getParam("root_frame", root_frame_))
     {
-        ROS_ERROR("Parameter 'target_frame' not set");
+        ROS_ERROR("Parameter 'reference_frame' not set");
         return false;
     }
 
-    if (nh_private.hasParam("update_rate"))
-    {    nh_private.getParam("update_rate", update_rate_);    }
-    else
-    {    update_rate_ = 68.0;    }    //hz
-
+    if(!nh_.getParam("update_rate", update_rate_))
+    {
+        update_rate_ = 50.0;    //hz
+    }
 
-    /// Cartesian Nodehandle
-    if (!nh_.getParam("chain_tip_link", chain_tip_link_))
+    /// Private Nodehandle
+    if(!nh_private.getParam("target_frame", target_frame_))
     {
-        ROS_ERROR("Parameter 'chain_tip_link' not set");
-        return false;
+        ROS_WARN("Parameter 'target_frame' not set. Using default 'cartesian_target'");
+        target_frame_ = "cartesian_target";
```

Comment:
Centralize by using a member or static const string outside: Would be easily changeable and usable in other locations.